pkg/workload/histogram/exporter/exporter_test.go

```diff
@@ -0,0 +1,151 @@
+		assert.Equal(t, "file path must end with .json", err.Error())
+	})
+
+	t.Run("Init", func(t *testing.T) {
+		file, err := os.CreateTemp("", "metrics.json")
+		require.NoError(t, err)
+		defer removeFile(file.Name())
+
+		writer := io.Writer(file)
+		err = exporter.Init(&writer)
+		require.NoError(t, err)
+	})
+
+	t.Run("SnapshotAndWrite", func(t *testing.T) {
+		file, err := os.CreateTemp("", "metrics.json")
+		require.NoError(t, err)
+		defer removeFile(file.Name())
+
+		writer := io.Writer(file)
+		err = exporter.Init(&writer)
+		require.NoError(t, err)
+
+		start := time.Time{}
+		elapsed := time.Second / 2
+		name := mockHistogramName
+		mockHistogram := hdrhistogram.New(0, 100, 1)
+		err = exporter.SnapshotAndWrite(mockHistogram, start, elapsed, &name)
+		require.NoError(t, err)
+		err = exporter.Close(nil)
+		require.NoError(t, err)
+
+		var data map[string]interface{}
+		buf, err := os.ReadFile(file.Name())
+		require.NoError(t, err)
+		err = json.Unmarshal(buf, &data)
+		assert.NoError(t, err)
+		assert.Equal(t, "mock_histogram", data["Name"])
+
+	})
+
+	t.Run("Close", func(t *testing.T) {
+		file, err := os.CreateTemp("", "metrics.json")
+		require.NoError(t, err)
+		defer removeFile(file.Name())
+
+		writer := io.Writer(file)
+		err = exporter.Init(&writer)
+		require.NoError(t, err)
+
+		err = exporter.Close(nil)
+		assert.NoError(t, err)
+	})
+}
+
+func TestOpenmetricsExporter(t *testing.T) {
+	exporter := &OpenmetricsExporter{}
+	t.Run("Validate", func(t *testing.T) {
+		err := exporter.Validate("metrics.txt")
+		assert.NoError(t, err)
+
+		err = exporter.Validate("metrics.json")
+		assert.Error(t, err)
+		assert.Equal(t, "file path must not end with .json", err.Error())
+	})
+
+	t.Run("Init", func(t *testing.T) {
+		file, err := os.CreateTemp("", "metrics.txt")
+		require.NoError(t, err)
+		defer removeFile(file.Name())
+
+		writer := io.Writer(file)
+		err = exporter.Init(&writer)
+		require.NoError(t, err)
+		assert.NotNil(t, exporter.writer)
+	})
+
+	t.Run("SnapshotAndWrite", func(t *testing.T) {
+		file, err := os.CreateTemp("", "metrics.txt")
+		require.NoError(t, err)
+		defer removeFile(file.Name())
+
+		writer := io.Writer(file)
+		err = exporter.Init(&writer)
+		require.NoError(t, err)
+
+		start := time.Time{}
+		elapsed := time.Second / 2
+		name := mockHistogramName
+		mockHistogram := hdrhistogram.New(0, 100, 1)
+		err = exporter.SnapshotAndWrite(mockHistogram, start, elapsed, &name)
+		require.NoError(t, err)
+		err = exporter.Close(nil)
+		require.NoError(t, err)
+
+		buf, _ := os.ReadFile(file.Name())
+		assert.Contains(t, string(buf), "# TYPE")
+	})
+
+	t.Run("Close", func(t *testing.T) {
+		file, err := os.CreateTemp("", "metrics.txt")
+		require.NoError(t, err)
+		defer removeFile(file.Name())
+
+		writer := io.Writer(file)
+		err = exporter.Init(&writer)
+		require.NoError(t, err)
+
+		err = exporter.Close(nil)
+		assert.NoError(t, err)
+	})
+}
+
+func removeFile(file string) {
+	_ = os.Remove(file)
```

Comment:
Nit: wrapping inside `assert.NoError` would alert us of any IO-related issues in CI, by failing the test. Otherwise, theoretically this could end up a silent "flake".